src/com/t_oster/liblasercut/drivers/GenericGcodeDriver.java

```diff
@@ -651,6 +659,45 @@ public void sendJob(LaserJob job, ProgressListener pl, List<String> warnings) th
     pl.taskChanged(this, "sent.");
     pl.progressChanged(this, 100);
   }
+
+@Override
+public void saveJob(LaserJob job) throws IllegalJobException, Exception {
+	checkJob(job);
+
+	String timestamp = new SimpleDateFormat( "yyyyMMddhhmmssSSS" ).format( new Date( ) );
+	String filename = "output_" + timestamp + ".gcode";
+	System.out.println("Creating file " + filename);
+	this.out = new PrintStream(new File(filename));
```

Comment:
We cannot assume, that the current directory is writeable. I think there is some getTemporaryFilename mehtod in IO or somehting like that